pkg/services/accesscontrol/eval/all_test.go

```diff
@@ -0,0 +1,109 @@
+package eval
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAll_Evaluate(t *testing.T) {
+	tests := []evaluateTestCase{
+		{
+			desc: "should return true for one that matches",
+			evaluator: All(
+				Permission("settings:write", Combine("settings", ScopeAll)),
+			),
+			permissions: map[string]map[string]struct{}{
+				"settings:write": {"settings:**": struct{}{}},
```

Comment:
I think, this test might fail if we end up fixing #36055 and grafana/grafana-enterprise#1651 by adding exact match for scopes. However, we can leave it as is as it works fine with the current implementation and does make sense conceptually. WDYT?

Reply:
same in eval_test.go

Reply:
Yes, if we implement exact match we should update the test in that pr